### PR TITLE
Remove 'no application' option from Add Source Wizard

### PR DESCRIPTION
**Pull request by CodyWMitchell**
# [RHCLOUD-24803](https://issues.redhat.com/browse/RHCLOUD-24803)

- Remove the option to choose 'No Application' in the 'Add a cloud source' wizard.
- Make selection of an application required.

**Before:**
image

**After:**
image

### CODE REVIEW

**Comment by CodyWMitchell** [comment]
Still seeing a failing test at in `addSourceSchema.js`. Now that choosing an application is required, the tests might take modifications to enter additional required data.

https://github.com/RedHatInsights/sources-ui/pull/1197/files#diff-664e62c8f8473189d06262d8e13daef004c8cbbb151d5271c57f55ba3db9909cR223

File: src/test/addSourceWizard/addSourceWizard/addSourceSchema.test.js
Comment (fhlavac) [inline src/test/addSourceWizard/addSourceWizard/addSourceSchema.test.js:1]: NO_APPLICATION_VALUE is still being imported but unused

File: src/test/addSourceWizard/addSourceWizard/superKey/applicationsStep.test.js
Comment (fhlavac) [inline src/test/addSourceWizard/addSourceWizard/superKey/applicationsStep.test.js:1]: NO_APPLICATION_VALUE is still being imported but unused

File: src/components/addSourceWizard/compileAllApplicationComboOptions.js [hunk 1]
```diff
@@ -54,23 +47,12 @@ export const labelMapper = (type, intl, hcsEnrolled) =>
     ),
   })[type.name];
 
-export const compileAllApplicationComboOptions = (applicationTypes, intl, activeCategory, hcsEnrolled) => [
+export const compileAllApplicationComboOptions = (applicationTypes, intl, hcsEnrolled) => [
```
Comment (fhlavac) [inline src/components/addSourceWizard/compileAllApplicationComboOptions.js:50]: the params change needs to be reflected across all usages of this function for the tests to pass

**Review by fhlavac — APPROVED** [review]

**Pull request merged**